cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/template/etl/validator/CoreValidator.java

```diff
@@ -0,0 +1,311 @@
+import org.apache.commons.validator.routines.DomainValidator;
+import org.apache.commons.validator.routines.DoubleValidator;
+import org.apache.commons.validator.routines.EmailValidator;
+import org.apache.commons.validator.routines.FloatValidator;
+import org.apache.commons.validator.routines.ISBNValidator;
+import org.apache.commons.validator.routines.InetAddressValidator;
+import org.apache.commons.validator.routines.IntegerValidator;
+import org.apache.commons.validator.routines.LongValidator;
+import org.apache.commons.validator.routines.RegexValidator;
+import org.apache.commons.validator.routines.ShortValidator;
+import org.apache.commons.validator.routines.UrlValidator;
+
+/**
+ * Utility class containing validator functions provided by org.apache.commons.validator.routines package.
+ */
+@Plugin(type = "validator")
+@Name("core")
+@Description("Core Validator functions.")
+public class CoreValidator implements Validator {
+
+  private static final EmailValidator EMAIL_VALIDATOR = EmailValidator.getInstance();
+  private static final DateValidator DATE_VALIDATOR = new DateValidator();
+  private static final CreditCardValidator CREDIT_CARD_VALIDATOR = new CreditCardValidator();
+  private static final DoubleValidator DOUBLE_VALIDATOR = new DoubleValidator();
+  private static final IntegerValidator INTEGER_VALIDATOR = new IntegerValidator();
+  private static final FloatValidator FLOAT_VALIDATOR = new FloatValidator();
+  private static final ShortValidator SHORT_VALIDATOR = new ShortValidator();
+  private static final LongValidator LONG_VALIDATOR  = new LongValidator();
+  private static final UrlValidator URL_VALIDATOR  = new UrlValidator();
+  private static final DomainValidator DOMAIN_VALIDATOR = DomainValidator.getInstance();
+  private static final ISBNValidator ISBN_VALIDATOR = new ISBNValidator();
+  private static final InetAddressValidator INET_ADDRESS_VALIDATOR = new InetAddressValidator();
+
+  @Override
+  public String getValidatorName() {
+    return "coreValidator";
+  }
+
+  @Override
+  public Object getValidator() {
+    return new ValidatorUtil();
+  }
+
+  /**
+   * Wrapper class for apache validator functions.
+   */
+  public static final class ValidatorUtil {
+
+    /**
+     * return true if the passed param is a valid date.
+     * uses {@link org.apache.commons.validator.routines.DateValidator} to check validity.
+     * @return boolean
+     */
+    public static boolean isDate(String date) {
+      return DATE_VALIDATOR.isValid(date);
+    }
+
+    /**
+     * return true if the passed param is a valid CreditCard.
+     * uses {@link org.apache.commons.validator.routines.CreditCardValidator} to check validity.
+     * @return boolean
+     */
+    public static boolean isCreditCard(String creditCard) {
+      return CREDIT_CARD_VALIDATOR.isValid(creditCard);
+    }
+
+    /**
+     * Checks if the field is null and length of the field is greater than zero not including whitespace.
```

Comment:
Checks if the input field is null, and if the length of the field is greater than zero, excluding whitespace.